packages/test/snapshots/src/snapshotStorageService.ts

```diff
@@ -0,0 +1,108 @@
+ *   callback should be provided when creating the storage service.
+ */
+export class SnapshotStorageService extends FileSnapshotReader implements IDocumentStorageService {
+    public constructor(
+        json: IFileSnapshot,
+        private readonly snapshotCb: (snapshot: IFileSnapshot) => void,
+    ) {
+        super(json);
+    }
+
+    public async uploadSummaryWithContext(summary: ISummaryTree, context: ISummaryContext): Promise<string> {
+        return Promise.reject(new Error("Invalid operation"));
+    }
+
+    public async createBlob(file: ArrayBufferLike): Promise<ICreateBlobResponse> {
+        return Promise.reject(new Error("Invalid operation"));
+    }
+
+    public async readBlob(blobId) {
+        return Promise.reject(new Error("Invalid operation"));
+    }
+
+    public async downloadSummary(handle: ISummaryHandle): Promise<ISummaryTree> {
+        return Promise.reject(new Error("Invalid operation"));
+    }
+
+    public get repositoryUrl(): string {
+        throw new Error("Invalid operation");
+    }
+
+    public async write(
+        tree: ITree,
+        parents: string[],
+        message: string,
+        ref: string,
+    ): Promise<IVersion> {
+        let dataStoreName = ref ? ref : "container";
+        if (tree && tree.entries) {
+            tree.entries.forEach((entry) => {
+                if (entry.path === ".component" && entry.type === TreeEntry.Blob) {
+                    const content = entry.value.contents.split(":");
+                    if (content[0] === `{"pkg"`) {
+                        dataStoreName = content[1].substring(1, content[1].lastIndexOf(`"`));
+                    }
+                }
+            });
+        }
+
+        // Remove null ids from the tree before calling the callback to notify the new snapshot. This is requried
+        // because the saved reference snapshots have the null ids removed.
+        removeNullTreIds(tree);
+
+        const commitName = `commit_${dataStoreName}`;
+        if (ref) {
+            this.commits[commitName] = tree;
+        } else {
+            this.docTree = buildSnapshotTree(tree.entries, this.blobs);
+            const fileSnapshot: IFileSnapshot = { tree, commits: {} };
+            // Call the callback with the snapshot in `IFileSnapshot` format.
+            this.snapshotCb(fileSnapshot);
+        }
+
+        return {
+            id: commitName,
+            date: new Date().toUTCString(),
+            treeId: FileSnapshotReader.FileStorageVersionTreeId,
+        };
+    }
+}
+
+/**
+ * Removed null ids from the snapshot tree for ease of reading and comparison.
+ */
+function removeNullTreIds(tree: ITree) {
+    for (const node of tree.entries) {
+        if (node.type === TreeEntry.Tree) {
+            removeNullTreIds(node.value);
+        }
+    }
+    // eslint-disable-next-line no-null/no-null
+    assert(tree.id === undefined || tree.id === null);
```

Comment:
shouldn't this be an if rather than an assert? otherwise this will try to delete all ids and fail on ids that are not null or undefined

Reply:
Yeah, that makes sense. I guess none of the trees in these tests have `id` so this works. I will update it.
FYI, it's the same here - https://github.com/microsoft/FluidFramework/blob/main/packages/drivers/file-driver/src/fileDocumentStorageService.ts. Not sure I want to change it here thought.

Reply:
or we could change the name to remove all tree ids or something, and keep the assert if the expectation is that they are always null